libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -538,7 +538,7 @@ string ABIFunctions::abiEncodingFunctionSimpleArray(
 						mstore(pos, sub(tail, headStart))
 						tail := <encodeToMemoryFun>(<arrayElementAccess>, tail)
 						srcPtr := <nextArrayElement>(srcPtr)
-						pos := add(pos, <elementEncodedSize>)
```

Comment:
This failed here because we only encode a pointer and not the actual data.

Reply:
`dynamicBase` says whether it is a fixed-length or dynamic-length?

Reply:
whether it is encoded statically or dynamically